File: app/src/main/java/com/isport/brandapp/wu/adapter/PractiseAdapter.java
package com.isport.brandapp.wu.adapter;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.isport.blelibrary.utils.Logger;
import com.isport.brandapp.R;
import com.isport.brandapp.wu.bean.ExerciseInfo;
import com.isport.brandapp.wu.bean.PractiseRecordInfo;

import java.util.ArrayList;
import java.util.List;

public class PractiseAdapter extends RecyclerView.Adapter<PractiseAdapter.MyViewHolder> {

    private List<PractiseRecordInfo> mDatas;
    private Context mContext;

    public PractiseAdapter(Context context, List<PractiseRecordInfo> data) {
        mContext = context;
        this.mDatas = data;
        if (data == null) {
            mDatas = new ArrayList<>();
        }

    }

    public void replaceData(List<PractiseRecordInfo> data) {
        this.mDatas = data;
        notifyDataSetChanged();
    }

    @Override
    public PractiseAdapter.MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MyViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_practise_record, parent, false));
    }

    @Override
    public void onBindViewHolder(PractiseAdapter.MyViewHolder holder, int position) {
        holder.tv_date.setText(mDatas.get(position).getKey());
        setItemRecyclerView(holder.recyclerview_itme, mDatas.get(position).getData());


        //实现点击效果

    }

    //私有属性
    private OnPractiseClickListener onItemClickListener = null;

    //setter方法
    public void setItemClickListener(OnPractiseClickListener onItemClickListener) {

        Logger.myLog("PractiseItemAdpter:onItemClickListener3 setItemClickListener"+onItemClickListener);
        this.onItemClickListener = onItemClickListener;
    }

    //回调接口
    public interface OnPractiseClickListener {
        void onItemClick(View v, ExerciseInfo note, int position);
    }

    @Override
    public int getItemCount() {
        return mDatas.size();
    }

    private void setItemRecyclerView(RecyclerView recyclerView, List<ExerciseInfo> mExerciseInfos) {
        recyclerView.setLayoutManager(new LinearLayoutManager(mContext, LinearLayoutManager.VERTICAL, false));
        PractiseItemAdapter practiseItemAdapter = new PractiseItemAdapter(mContext, mExerciseInfos);
        practiseItemAdapter.setOnItemClickListener(new PractiseItemAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View v, ExerciseInfo note, int position) {
                Logger.myLog("PractiseItemAdpter:onItemClickListener2" + onItemClickListener);
                if (onItemClickListener != null) {
                    onItemClickListener.onItemClick(v, note, position);
                }
            }
        });
        recyclerView.setAdapter(practiseItemAdapter);

    }

    class MyViewHolder extends RecyclerView.ViewHolder {
        RecyclerView recyclerview_itme;
        private TextView tv_date;
        LinearLayout layout;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            recyclerview_itme = itemView.findViewById(R.id.recyclerview_itme);
            tv_date = itemView.findViewById(R.id.tv_date);
            layout = itemView.findViewById(R.id.layout);

        }
    }
}

File: app/src/main/java/com/isport/brandapp/wu/adapter/PractiseAdapter.kt
package com.jkcq.homebike.history.adpter

import android.content.Intent
import android.graphics.Color
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import brandapp.isport.com.basicres.commonutil.UIUtils
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.isport.brandapp.R
import com.isport.brandapp.wu.activity.PractiseDetailActivity
import com.isport.brandapp.wu.adapter.PractiseItemAdapter
import com.isport.brandapp.wu.bean.ExerciseInfo
import com.isport.brandapp.wu.bean.PractiseRecordInfo


/**
 *  Created by BeyondWorlds
 *  on 2020/6/30
 */
class PractiseAdapter(data: MutableList<PractiseRecordInfo>) :
        BaseQuickAdapter<PractiseRecordInfo, BaseViewHolder>(R.layout.item_practise_record, data) {
    init {

        addChildClickViewIds(R.id.layout_head)
    }


    private fun updateRecyclerView(recyclerView: RecyclerView) {


    }

    lateinit var historyDayDetailAdapter: PractiseItemAdapter

    private fun setItemRecyclerView(recyclerView: RecyclerView, mExerciseInfos: MutableList<ExerciseInfo>) {
        recyclerView.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        historyDayDetailAdapter = PractiseItemAdapter(context, mExerciseInfos)
        historyDayDetailAdapter.setOnItemClickListener { v, note, position ->


            val intent = Intent(context, PractiseDetailActivity::class.java)
            intent.putExtra("info", note)
            context.startActivity(intent)


            // intent.putExtra("url", AppConfiguration.ropedetailurl + "?userId=" + TokenUtil.getInstance().getPeopleIdInt(BaseApp.instance) + "&ropeId=" + mExerciseInfos.get(position).ropeSportDetailId)
        }
        recyclerView.adapter = historyDayDetailAdapter
    }

    override fun convert(holder: BaseViewHolder, item: PractiseRecordInfo) {
        var rec = holder.getView<RecyclerView>(R.id.rec_day)


        holder.setText(R.id.tv_date, "" + item.key)
        holder.setText(R.id.tv_sum_count, String.format(UIUtils.getString(R.string.total_practise_times), item.data.size))

        if (item.isSelect) {
            holder.setVisible(R.id.rec_day, true)
            if (item!!.data != null) {
                item!!.currentShowList.addAll(item!!.data)
                holder.setImageResource(R.id.iv_arrow_right, R.drawable.icon_rope_up)
                holder.setBackgroundResource(R.id.layout_view, R.drawable.shape_btn_white_20_bg)
                setItemRecyclerView(rec, item!!.currentShowList)
            } else {
                holder.setImageResource(R.id.iv_arrow_right, R.drawable.icon_rope_down)
                holder.setBackgroundColor(R.id.layout_view, Color.TRANSPARENT)
            }
        } else {
            if (item!!.currentShowList != null) {
                item!!.currentShowList.clear()
                setItemRecyclerView(rec, item!!.currentShowList)
            }
            holder.setVisible(R.id.rec_day, false)
            holder.setImageResource(R.id.iv_arrow_right, R.drawable.icon_rope_down)
            holder.setBackgroundColor(R.id.layout_view, Color.TRANSPARENT)
        }

    }
}